SimGeneral/MixingModule/python/pixelDigitizer_cfi.py

```diff
@@ -9,3 +10,7 @@
     makeDigiSimLinks = cms.untracked.bool(True)
 )
 
+from Configuration.StandardSequences.Eras import eras
+eras.phase2_tracker.toModify( pixelDigitizer, 
+                              pixel = phase2TrackerDigitizer)
```

Comment:
IIUC you want
```
eras.phase2_tracker.toReplaceWith(pixelDigitizer, phase2TrackerDigitizer)
```
It is the `pixelDigitizer` that ends up as `process.mix.digitizers.pixel`.

After this I'd suggest to change
```
from SimTracker.SiPhase2Digitizer.phase2TrackerDigitizer_cfi import phase2TrackerDigitizer as _phase2TrackerDigitizer
```
to not pollute the module scope with unnecessary stuff (that may later end up in `process`).